storage-example.yml

```
@@ -0,0 +1,113 @@
+# example storage.yml
+
+##### general structure:
+
+# the name of storage item
+my-thing:
+  type: X   # options: s3, s3-policy, redis, opensearch, rds-pg, aurora-pg
+
+  environments:  # per environment configuration is defined here.
+    default:   # the optional default key is used to apply a default value to all environments
+      some-setting: x
+      another-setting: y
+
+      plan: a-valid-plan  # some storage types support plans/t-shirt sizes.  These are listed in storage-plans.yaml in this repo.
+
+    production:  # you can override defauls on a per env basis
+
+      plan: some-other-plan  # Any of the keys defined in a storage plan can also be overridden
+      volume-size: 200
+
+
+##### s3 bucket example:
+
+my-s3-bucket:
+  # currently no other features are supported such as bucket versioning, etc.  If you need additional features, you can use the s3-policy type to give a service access to an existing bucket.
+
+  type: s3   # creates an s3 bucket in each environment and gives the listed services permissions to access the bucket
+
+  readonly: true  # services are granted read only access to the bucket
+
+  services:  # services that require access to the bucket.
+    - "web"
+    - "web-celery"
+
+  environments:
+    # NOTE: [bucket names must be unique across all AWS accounts in the AWS (Standard Regions) Partition](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html) and need to be defined for each environment
```

@yusufsheiqh made a good point about this not being markdown. Sorry...
```suggestion
    # NOTE: bucket names must be unique across all AWS accounts in the AWS (Standard Regions) Partition (see https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html) and need to be defined for each environment
```

Will approve now anyway as this is not a deal breaker.